src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibLeaseCoordinator.java

```diff
@@ -292,6 +292,20 @@ public Checkpoint getCheckpointObject(String shardId) throws KinesisClientLibExc
         }
     }
 
+    public Set<String> getChildShardIds(String shardId) {
+        String errorMessage = "Unable to fetch childShardIds for shardId " + shardId;
+        try {
+            KinesisClientLease lease = leaseManager.getLease(shardId);
+            if (lease == null) {
+                throw new KinesisClientLibIOException(errorMessage);
```

Comment:
I agree that this exception throw is not necessary because if a shard doesn't reach the shard end, the childShardIds entry would be empty. As we are already validating the childShardIds and throw exception in advanceIteratorTo in KinesisDataFetcher, I will remove this exception.

Reply:
If leases could potentially be null, won't we get NPEs when trying to retrieve their child shards? I think this protection is still needed